File: TOI-Zero_68/A2/001/001.py
n, m = map(int, input().split())

red = [0] + list(map(int, input().split()))
blue = [0] + list(map(int, input().split()))

count = 0

for i in range(n):
    for j in range(m):
        if (i % 2) == (j % 2):
            cond = (red[i] < blue[j] and red[i+1] > blue[j+1]) or \
                   (red[i] > blue[j] and red[i+1] < blue[j+1])
        else:
            cond = (red[i] < blue[j+1] and red[i+1] > blue[j]) or \
                   (red[i+1] > blue[j] and red[i] < blue[j+1])
        count += cond

for i in range(n + 1):
    start = i % 2
    for j in range(start, m + 1, 2):
        if red[i] == blue[j]:
            count += 1

print(count)

File: TOI-Zero_68/A2/001/001.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n,m;
	cin >> n >> m;
	ll red[n+1],blue[m+1];
	red[0]=0,blue[0]=0;
	for(int i=1;i<=n;i++){
		cin >> red[i];
	}
	for(int i=1;i<=m;i++){
		cin >> blue[i];
	}
	ll count=0;
	for(int i=0;i<n;i++){
		for(int j=0;j<m;j++){
			if(i%2==j%2){
				count += ((red[i]<blue[j] && red[i+1]>blue[j+1]) || (red[i]>blue[j] && red[i+1]<blue[j+1]));
			}
			else{
				count += ((red[i]<blue[j+1] && red[i+1]>blue[j]) || (red[i+1]>blue[j] && red[i]<blue[j+1]));
			}
		}
	}
	for(int i=0;i<=n;i++){
		for(int j=(i%2);j<=m;j+=2){
			count += red[i]==blue[j];
		}
	}
	cout << count;
}
